src/pages/html/attributes/required/index.md

```diff
@@ -1,15 +1,30 @@
 ---
 title: Required
 ---
-## Required
+## Href
 
-This is a stub. <a href='https://github.com/freecodecamp/guides/tree/master/src/pages/html/attributes/required/index.md' target='_blank' rel='nofollow'>Help our community expand it</a>.
+The `required` attribute is a boolean attribute. The following syntax is used: `<input required>`. If the `required` attribute is used in an `<input>` element, that input field is mandatory to fill before submitting the form. The `required` attribute must be used with the following input types: 
+* `text` 
+* `password` 
+* `email`
+* `tel` 
+* `url` 
+* `search`
+* `date pickers` 
+* `number`
+* `checkbox`
+* `radio` 
+* `file`
 
-<a href='https://github.com/freecodecamp/guides/blob/master/README.md' target='_blank' rel='nofollow'>This quick style guide will help ensure your pull request gets accepted</a>.
+### Examples:
 
-<!-- The article goes here, in GitHub-flavored Markdown. Feel free to add YouTube videos, images, and CodePen/JSBin embeds  -->
+``` 
```

Comment:
Please use `html` syntax highlighting, you can checkout [this part](https://github.com/freeCodeCamp/guides/blob/master/CONTRIBUTING.md#code-blocks) of the Article Style Guide.